### PR TITLE
=str - #17827 - Introduces  operator for Sources and Flows, to attach…

### PR DESCRIPTION
**Pull request by viktorklang**
Is this the inline broadcast you've been looking for?

### CODE REVIEW

**Comment by viktorklang** [comment]
TlsEndpointVerificationSpec failed. Unrelated to this PR.

**Comment by drewhk** [comment]
That failure is being fixed,and yes, it is not related.

**Comment by ktoso** [comment]
PLS BUILD

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala [hunk 1]
```diff
@@ -214,6 +216,25 @@ class GraphBroadcastSpec extends AkkaSpec {
       upsub.expectCancellation()
     }
 
+    "alsoTo must broadcast" in assertAllStagesStopped {
+      val p, p2 = TestSink.probe[Int](system)
+      val (ps1, ps2) = Source(1 to 6).alsoToMat(p)(Keep.right).toMat(p2)(Keep.both).run()
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: nice

how does the `Mat` things look like if I would also care about the materialized value of the `Source`, e.g. `Source.actorRef`?

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala [hunk 2]
```diff
@@ -214,6 +216,25 @@ class GraphBroadcastSpec extends AkkaSpec {
...
+
+    "alsoTo must continue if sink cancels" in assertAllStagesStopped {
+      val p, p2 = TestSink.probe[Int](system)
+      val (ps1, ps2) = Source(1 to 6).alsoToMat(p)(Keep.right).toMat(p2)(Keep.both).run()
+      ps2.request(6)
+      ps1.cancel()
+      ps2.expectNext(1, 2, 3, 4, 5, 6)
+      ps2.expectComplete()
+    }
   }
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: add a few more test scenarios?

**Comment by patriknw** [comment]
reference documentation?

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala (hunk 1 above)
Comment (viktorklang) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: I assume you mean if "Keep.right" is "Keep.both" in the example? I assume it'll return: `val ((ps0, ps1), ps2)`

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala (hunk 2 above)
Comment (viktorklang) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: Something more specific? (I don't want to duplicate all broadcast tests)

**Comment by viktorklang** [comment]
@patriknw If this suggested change is desired, then the creation of reference documentation is definitely planned.

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala (hunk 2 above)
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: I wrote this before I saw that the implementation reused the other broadcast. good as is

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala (hunk 1 above)
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: thanks

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala (hunk 2 above)
Comment (viktorklang) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: :+1:

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala (hunk 1 above)
Comment (viktorklang) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBroadcastSpec.scala]: Np! :)

**Comment by rkuhn** [comment]
LGTM: please wrap up and merge at your convenience :-) Thanks!

**Pull request merged**